lib/Dialect/TTIR/IR/TTIROps.cpp

```diff
@@ -300,6 +300,54 @@ ::mlir::LogicalResult mlir::tt::ttir::ReshapeOp::verify() {
   return success();
 }
 
+::mlir::LogicalResult mlir::tt::ttir::SliceOp::verify() {
+  ::mlir::RankedTensorType inputType = getInput().getType();
+  ::mlir::ArrayAttr begins = getBeginsAttr();
+  ::mlir::ArrayAttr ends = getEndsAttr();
+  ::mlir::ArrayAttr step = getStepAttr();
+  ::mlir::RankedTensorType outputType = getOutput().getType();
+
+  // Verify that the input is at least 1D tensor
+  if (inputType.getRank() < 1) {
+    return emitOpError("Input must be at least a 1D tensor");
+  }
+
+  // Verify that the input rank matches number of elements in begins, ends, and step
+  size_t input_rank = static_cast<size_t>(inputType.getRank());
+  if (input_rank != begins.size() || input_rank != ends.size() || 
+      input_rank != step.size()) {
+    return emitOpError("Begins, ends, and step attributes must have the same "
+                       "number of elements as the input tensor rank");
+  }
+
+  // Validate that the output tensor has the same element type as the input tensor
+  if (inputType.getElementType() != outputType.getElementType()) {
+    return emitOpError("Output tensor must have the same element type as the input tensor");
+  }
+
+  // Verify the output tensor rank
+  if (inputType.getRank() != outputType.getRank()) {
+    return emitOpError("Output tensor must have the same rank as the input tensor");
+  }
+
+  // Verify the output tensor dimensions based on begins, ends, and step
+  for (size_t i = 0; i < input_rank; ++i) {
+    int32_t begin = ::mlir::cast<::mlir::IntegerAttr>(begins[i]).getInt();
+    int32_t end = ::mlir::cast<::mlir::IntegerAttr>(ends[i]).getInt();
+    int32_t stride = ::mlir::cast<::mlir::IntegerAttr>(step[i]).getInt();
+
+    // Calculate the expected size of the output dimension
```

Comment:
Additionally, let's check that `begins` and `ends` values for each dimension exist in input tensor.

E.g.
```
a = Tensor([1, 2, 3]);  // 1d tensor
begins = [0];
ends = [5];
step = [2];
b = ttnn::slice(a, begins, ends, step);
```

This will pass the `expectedDimSize` check as `(5-0+2-1) / 2 = 3` but should fail on the proposed check.